### PR TITLE
handles return value from methods

### PR DESCRIPTION
**Pull request by siavash1986**
replaced File.delete() with FileUtils.deleteAndReport().
It's a bad practice to ignore return values and this commit fixes this issue in the code base.
`UtilFiles.deleteAndReport()` gives a file a second chance to delete by calling `File.deleteOnExit()` and report the result. 
Also this commit will log the return value of File.renameTo() and File.mkdirs() if they fail

#### What has been done to verify that this works as intended?
ran all tests and also simulated on the Android Emulator.

#### Are there any risks to merging this code? If so, what are they?
It's safe

### CODE REVIEW

**Comment by lognaturel** [comment]
`deleteOnExit` has [the following warning](https://developer.android.com/reference/java/io/File.html#deleteOnExit()): "Note that on Android, the application lifecycle does not include VM termination, so calling this method will not ensure that files are deleted". It seems like the `deleteAndReport` method may need some rethinking. What are our options there?

**Comment by lognaturel** [comment]
This is a kind of change I'm not really sure what to do with and I'd love your thoughts @jknightco. It initially seems like using something like `UtilFiles.deleteAndReport()` would be an improvement but the method doesn't actually do what is intended because of what I mentioned in my previous comment. One could say it's still a good change because it gives an opportunity to improve the deletion in one place but it introduces more static calls and I'm not entirely sure that wrapping the deletion in a method is the right way to go. Using `finally` might be more appropriate.

**Comment by heyjamesknight** [comment]
tl;dr: remove deleteOnExit and move to non-static method and I think its an improvement.

Sorry, @lognaturel and I talked about this offline and I never responded here. I'm not a fan of code that doesn't do what it looks like it does—considering `deleteOnExit` is not what it seems I don't think we should use it. 

I do, however, like removing File.delete(), but not into _another_ static method: if we can move it into `FileUtil` as a normal method and make sure we have an instance of FileUtil where we need it then it'll make it easy to mock in tests.

**Comment by lognaturel** [comment]
Thanks @jknightco, that all sounds really good.

@siavash1986 I'm going to close this for now but please resubmit with the changes suggested! If you have any questions about those you can ask here or find us in the #collect-code Slack channel (http://slack.opendatakit.org)
